### DISCUSSION TITLE
Android BackButton Always Navigates Back

### DISCUSSION THREAD

**Discussion by jmichas** [PrismLibrary/Prism#2831] [.NET MAUI]
In Xam Forms I could prevent the back button on a per page basis by simply overriding the `OnBackButtonPressed()` method in a page's code behind.

It seems that this no longer works with MAUI Prism. My page gets navigated back and I assume it is because the `PrismAppBuilder` sets the lifecycle event for `OnBackPressed` and the back button always causes the `navigation.GoBackAsync()` line.

Is there a way to override this and prevent any navigation? 

I can't implement my own version of the lifecycle event because the `PrismWindow` is internal.

And if I'm wrong in my assumption could you point me to where/how the page is navigating back?

Thanks.

**Comment by soroshsabz** [comment]
ITNOA

I think maybe https://github.com/PrismLibrary/Prism/issues/2815 related to this discussion.
